### ISSUE TITLE
'save' function fails

### ISSUE THREAD

**Issue by hongkongkiwi** [balderdashy/sails-redis#24]
When using waterline 'save' function on an instance it fails with error:
[Error: Primary key cannot be updated]

This happens even when I'm not updating a single field.

However when using the Update function on the Model it works fine.

Seems like a bug and should probably have a test for it, (get a record then save it)

**Comment by Ryanc1256** [comment]
Closing as it's apparently fixed. I'll re-open it if someone still has a problem.

**Comment by JerryC8080** [comment]
@Ryanc1256 
i also have the error with 'save' function, but my error is :     
`AdapterError: Record does not satisfy unique constraints`

my code like this:    
```javascript
Permission.findOne(permissionID).populate('roles').then(function (permission) {
  permission.roles.add(adminRole.id);
  return permission.save();
});
```

i has seen [#67](https://github.com/balderdashy/sails-redis/pull/67) and installing the `0.10.4` version, but still come across this error.

**Comment by Ryanc1256** [comment]
@JerryC8080 whats your Permission model look like if i may ask?

**Comment by JerryC8080** [comment]
@Ryanc1256 
Sure! 

model/Permission.js
```javascript
var uuid = require("uuid");

module.exports = {

  schema        : true,
  identity      : 'Permission',
  tableName     : 'permission',
  connection    : 'redisClient',

  attributes: {
    uuid            : {type: 'string', unique: true, size: 36},
    name          : {type: 'string', unique: true, required: true, size: 50},
    description  : {type: 'string', size: 255},
    action          : {type: 'string', index: true, notNull: true, size: 50, enum: ['create', 'read', 'update', 'delete']},
    relation        : {type: 'string', enum: [ 'role', 'owner' ], defaultsTo: 'role', index: true ,size:10},

    // association
    model       : {model: 'Model',required: true},
    roles         : {collection: 'Role', via: 'permissions', dominant: true}
  },

  beforeCreate: function (values, next) {
    values.uuid = uuid.v4();
    next();
  }
```

**Comment by Ryanc1256** [comment]
@JerryC8080 Thank's for that, i found it's a bug with the update method, for some reason it check's if the unique key's are already in use if so they will throw an error and not update the model. 

So I'm gonna have to fix that before you models can get updated properly.

Thanks for the find though!

**Comment by JerryC8080** [comment]
oh, it's my pleasure.
as you mean, am i just need to remove `unique` validation that will work normally in this version?
`sails-redis` will be used in my project with latest version , 
if possible i want to waiting your show....hahhhh
thanks a lot!

**Comment by Ryanc1256** [comment]
@JerryC8080 Yeah it should work fine and like normal, ill try a fix that unique key bug tonight after work, if i get some free time to work on it.

**Comment by Ryanc1256** [comment]
Okay it's going to be harder than i thought

**Comment by FabienHenon** [comment]
I also have this issue. Is it solved ?

**Comment by Ryanc1256** [comment]
@FabienHenon not yet, i've been looking on how to fix it but with my limited time, i just cant do it atm...

It's a problem with unique value's if you remove them it will fix them.

**Comment by FabienHenon** [comment]
Alright. Do you have a workaround?
Actually I can't just remove unique because it would break my validations :(

Fabien Henon
Tél. : +33 (0)6 99 43 61 46

**Comment by Ryanc1256** [comment]
@FabienHenon Not at the moment, unless you comment one part out in the sails-redis module i can tell you want lines to comment out, and it should go through... the only time it would be iffy is when you use the .save method and it updates the unique values.

**Comment by FabienHenon** [comment]
Ok. Could you tell Le those lines ?

Fabien Henon
Tél. : +33 (0)6 99 43 61 46

**Comment by Ryanc1256** [comment]
@FabienHenon woop's forgot to msg you back.... 
Comment this line out
https://github.com/balderdashy/sails-redis/blob/master/lib/database/index.js#L555
I think that should fix your problem until someone actually fixes it...

**Comment by FabienHenon** [comment]
Ok. Thank you

**Comment by sailsbot** [comment]
Thanks for posting, @hongkongkiwi.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by steverc** [comment]
Hi all,
any news on this issue? I'm facing the same problem and it seems still open...
Thanks

**Comment by Ryanc1256** [comment]
Hey @steverc 
Nope, still trying to find time to fix it...
But if you want to, you can go and fix it :p

**Comment by sailsbot** [comment]
Thanks for posting, @hongkongkiwi.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by johnabrams7** [comment]
@hongkongkiwi @sobtiankit @CWyrtzen @Nearhan @steverc @FabienHenon @Ryanc1256 @JerryC8080 @hunterboerner  Hey everyone, we're moving the sails-redis issues to the main Sails repo (balderdashy/sails/issues) for more exposure to the community.  I see this issue was re-opened for additional help, the extra exposure should help gather more input. Thanks!
